### PR TITLE
Remove `registries` config and `--registry` option

### PR DESCRIPTION
**Pull request by bitoku**
#### What type of PR is this?

/kind cleanup
/kind deprecation

#### What this PR does / why we need it:

This PR removes `registries` config and `--registry` option

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

See #8161 for the context.

#### Does this PR introduce a user-facing change?

```release-note
Remove `registries` config in `crio.image` and `--registry` option which have been already deprecated.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -421,7 +421,7 @@ completions-generation:
 
 docs: $(MANPAGES)
 
-docs-generation:
+docs-generation: bin/crio
```
Comment (saschagrunert) [inline Makefile:424]: It may be possible that we break consumers with that.

Comment (saschagrunert) [inline Makefile:424]: Let's revert that, it may break existing users.

**Review by saschagrunert — CHANGES_REQUESTED** [review]

**Comment by kwilczynski** [comment]
@cri-o/cri-o-maintainers, can you give this another pass? Thank you!

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**